cmds/webboot/webboot.go

```diff
@@ -227,6 +227,16 @@ func main() {
 			cacheDir = cachePath
 		}
 	}
+
+	if err := menu.Init(); err != nil {
+		log.Fatalf(err.Error())
+	}
+	defer menu.Close()
+
+	// Suppress output, else it might overlap with the menu
```

Comment:
I wonder if there's a way we could just defer the output to after the menu exists rather than discarding it in case the menu closes with an error and we want to see the log output?

Reply:
That's a good point. Maybe I could write it to a file?

Reply:
Rule 1: no discarding output
Rule 2: no writing it to random files. You can put it in memory.

you can do this for logging. 

r, w := io.Pipe
log.SetOutput(w)
the log output will appear in the pipe. you can get it later by reading from r.

also this os.Stdout? Why do this? never do this IOW and don't plan on using files ...
where is the code that writes to os.Stdout? A command you spawn? Let's talk some more. 
Let's do one thing at a time however. Can you fix the log bits first?

Reply:
I'll make that change for the log output. Should we do anything with `r` at the end?

Also, the only thing that gets written to `os.Stdout` is the download progress.

Reply:
I made the change locally, but the program hangs at the function call that generates the logs. I think the problem might have to do with the pipe being synchronous.

Would it be ok to make a Writer class with an internal buffer?

Reply:
use a bytes.Buffer instead of the io.Pipe. Sorry, I have you bad advice.

Reply:
No problem, it should be updated now.